ssh-notes/using-ssh-keys-with-passwords.md

```diff
@@ -0,0 +1,84 @@
+# Using SSH Keys with Passwords
+
+When you create an SSH key, you are allowed to assign a password to it.  This means whenever you attempt to use the key it will prompt you with a password.  Some software, such as VS Code, can deal with this for you and cache the appropriate permissions it needs.
+
+Other software (for example the command line version of git) will always prompt for the password associated with the SSH key being used.  This can be rather annoying.  It is possible to cache the permissions such that you only need to enter the password at the start of a session.
+
+This is done using a tool called **ssh-agent**.  It is available on both the Windows and WSL environments.  This document will focus primarily on the WSL environment.
+
+## SSH-Agent Under WSL
+
+### Overview
+
+The ssh-agent program will allow a password for a key to be cached for automatic reuse.  This means we need to perform the following tasks
+
+- Make sure that ssh-agent is running
+- Add the key to ssh-agent
+
+### Procedure
+
+To use ssh-agent to cache your key password, do the following
+
+1. Start a WSL shell.  Make sure you are in your home directory by running this command
+
+   ```text
+   cd ~
+   ```
+
+2. Amend your .bashrc script to load the ssh-agent program when you login to a WSL shell.  This can be accomplished by using a text editor.  A simple text editor to use is called 'nano'.  Once you are editing the .bashrc script, then add the following to the end of the script.
```

Comment:
You can use VS Code rather than another editor. For example by entering `code ~/.bashrc` in WSL. I think it would be preferable to avoid alternative editors. (Anyway, vi is king 😉).

Reply:
The only point it that the reader may not have VS Code up and running properly at this point?  I suggested nano as it is easy to use even if you have little experience with linux

Reply:
Amended as discussed